gcp/table_gcp_dns_record_set.go

```diff
@@ -0,0 +1,209 @@
+				Type:        proto.ColumnType_JSON,
+				Transform: transform.FromField("RecordSet.NullFields"),
+			},
+
+			// Steampipe standard columns
+			{
+				Name:        "title",
+				Description: ColumnDescriptionTitle,
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("RecordSet.Name"),
+			},
+			{
+				Name:        "akas",
+				Description: ColumnDescriptionAkas,
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getDnsRecordSetAka,
+				Transform:   transform.FromValue(),
+			},
+
+			// Standard gcp columns
+			{
+				Name:        "location",
+				Description: ColumnDescriptionLocation,
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromConstant("global"),
+			},
+			{
+				Name:        "project",
+				Description: ColumnDescriptionProject,
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     getProject,
+				Transform:   transform.FromValue(),
+			},
+		},
+	}
+}
+
+//// LIST FUNCTIONS
+
+func listDnsRecordSet(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	plugin.Logger(ctx).Trace("listDnsRecordSet")
+
+	// Get the details of Cloud DNS Managed Zone
+	managedZone := h.Item.(*dns.ManagedZone)
+
+	// Create Service Connection
+	service, err := DnsService(ctx, d)
+	if err != nil {
+		return nil, err
+	}
+
+	// Get Project details
+	projectData, err := activeProject(ctx, d)
+	if err != nil {
+		return nil, err
+	}
+	project := projectData.Project
+
+	resp, err := service.ResourceRecordSets.List(project, managedZone.Name).Do()
+	if err != nil {
+		return nil, err
+	}
+
+	for _, recordset := range resp.Rrsets {
+		d.StreamListItem(ctx, recordSetInfo{recordset, managedZone.Name})
+	}
+
+	return nil, nil
+}
+
+//// HYDRATE FUNCTIONS
+
+func getDnsRecordSet(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	plugin.Logger(ctx).Trace("getDnsRecordSet")
+
+	// Get the details of Cloud DNS Managed Zone
+	managedZone := h.Item.(*dns.ManagedZone)
+
+	// Create service connection
+	service, err := DnsService(ctx, d)
+	if err != nil {
+		return nil, err
+	}
+
+	// Get project details
+	projectData, err := activeProject(ctx, d)
+	if err != nil {
+		return nil, err
+	}
+	project := projectData.Project
+
+	name := d.KeyColumnQuals["name"].GetStringValue()
+	rrset_type := d.KeyColumnQuals["type"].GetStringValue()
+	managedZoneName := managedZone.Name
+
+	resp, err := service.ResourceRecordSets.List(project, managedZoneName).Name(name).Type(rrset_type).Do()
+	if err != nil {
+		return nil, err
+	}
+
+	if len(resp.Rrsets) < 1 {
+		return nil, nil
+	}
```

Comment:
@ParthaI Can you please add a comment to define why are we using this check?

Reply:
Actually we do not need the above line code so removed that code block